NTHU_Course/settings/default.py

```diff
@@ -57,6 +60,32 @@
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
 )
 
+
+AUTHENTICATION_BACKENDS = (
+    'social.backends.facebook.FacebookOAuth2',
+    'django.contrib.auth.backends.ModelBackend',
+)
+
+TEMPLATE_CONTEXT_PROCESSORS = (
+    'django.contrib.auth.context_processors.auth',
+    'social.apps.django_app.context_processors.backends',
+    'social.apps.django_app.context_processors.login_redirect',
+)
```

Comment:
It's just commented out since it has been added...
https://github.com/django/django/commit/1dc65cb6ab158fae3ccae55512a4f49ec3654758

I've read somewhere that they don't want to give templates too much information